### ISSUE TITLE
Alarm clock not working

### ISSUE THREAD

**Issue by DomeDrake** [orgzly-revived/orgzly-android-revived#49]
Thank you for all your efforts to keep Orgzly in developement!

I have encountered an issue, the alarm clock feature "Use alarm clock: For reminders with the time fo day set" doesn't work on my phone. (Android 12)

In both v1.8.10 and v1.8.11 the push notification appears as usual. Also in both versions I can see my alarm clock icon appearing in the android status bar, next to my wifi icon etc., when I set a time of day for the scheduled time of the task, meaning something is happening, but my actual alarm never rings.

This issue has already been mentioned in [Orgzly #654](https://github.com/orgzly/orgzly-android/issues/654) for Orgzly, however I wanted to mention it in the revived project as well.

Again, I'm incredibly happy to see you all do the work you're doing, thank you!

**Comment by appletalk** [comment]
Same issue here, on Android 14 (GrapheneOS).  I see a notification on my lock screen with the next upcoming alarm which is correct, however I only get the push notification at the time specified not an alarm.

**Comment by amberin** [comment]
Check the app info screen and go to the bottom. You should see the "alarms & reminders" permission there, make sure it's checked. I recently discovered that tests are failing on Android 14 because this permission is no longer granted by default. I will fix this by adding some permission checks in the code.

**Comment by DomeDrake** [comment]
On Android 12 (OxygenOS 12.1), when looking at the App permissions, I only see the "Storage" permission:

<img src="https://github.com/orgzly-revived/orgzly-android-revived/assets/149258400/454d07f2-6d2e-4461-b778-7e5d151acc1b" width="200" />

Looking at "all permissions" (by clicking on the three dots in the top right corner in the screenshot above), there is a permission called "android.permission.SCHEDULE_EXACT_ALARM":

<img src="https://github.com/orgzly-revived/orgzly-android-revived/assets/149258400/75356d25-e7aa-4026-9307-8d32b01ff7a3" width="200" />

Unfortunately, with the setup you can see in my screenshots, granting all possible permissions afaik, the alarm is still not successfully working, like before only the icon in the status bar appears (which is hidden in the screenshots).

If I can help by providing any more information, please do let me know! I'm ever so grateful for your continued work on Orgzly Revived! :)

**Comment by jamal4hm3d** [comment]
Hi,

Is there any update on this.

I am on android 13 and i am also facing same issue.

Can someone please help with this.

Thanks.

**Comment by amberin** [comment]
I wonder what exactly the intended behavior is and if this has ever worked - it doesn't seem that way from the original issue.

**Comment by jamal4hm3d** [comment]
Hi @amberin,

For me, Some times i will miss the notification. So, i just want a loud remaindrer that will stop once i close it.

I thought the option would do that but it is not working.

If possible, Please look into this.

I really appreciate this project.

Thanks.

**Comment by amberin** [comment]
Yes, I understand the intended use case, and it seems very useful. I'm just trying to understand if the current code has ever worked as intended, because I can't find any tests of this feature, so I'm not sure where to look for the bug(s) that is causing it to fail.
